azure-iot-device/azure/iot/device/common/pipeline/config.py

```diff
@@ -77,3 +82,23 @@ def _sanitize_cipher(cipher):
             raise TypeError("Invalid type for 'cipher'")
 
         return cipher
+
+    @staticmethod
+    def _rectify_keep_alive(keep_alive):
+        if keep_alive and not isinstance(keep_alive, (int, float)):
```

Comment:
>if keep_alive and not isinstance(keep_alive, (int, float)): [](start = 8, length = 59)

Why not just convert this to int, always? That would presumably get rid of the need for the builtins import as well

Reply:
do you mean just try to convert to int and throw an error if conversion goes wrong ? the built in int is only used to support long for python 2.7.  Long for python 3 falls within the int range but for python 2 if someone passes 9L it will fail if its just the normal int. 

---
In reply to: [461199086](https://github.com/Azure/azure-iot-sdk-python/pull/635#discussion_r461199086) [](ancestors = 461199086)